README.md

```diff
@@ -558,6 +559,23 @@ import {requestVibration} from '@tef-novum/webview-bridge';
 requestVibration('error');
 ```
 
+### matchContactData
+
+Returns contacts info given an array of phone numbers.
+
+```javascript
+matchContactData: (numbers: Array<string>) => Promise<{
+    matches: {
+        [key: string]: {
```

Comment:
what's this key? the phone number, an identifier? give it a name

Anyway, do we really need a map here? this could be just a collection

Reply:
this key is the normalized number, maybe yo prefer normalizedNumber instead of key.
As I've written in the prev comment, this is a TS done by NPI team

Reply:
I think it would be better to include it in the collection

Reply:
having a key makes sense when you want to pick items by key but in this case the key is unknown to the caller so it is useless

if you want to return a collection, use a collection (array of contacts). Keep it simple

Reply:
Unkown? this key is necessary to the caller and identify uniquely the information.

Reply:
The map is keyed by the numbers you request on the bridge. The map is nice to fastly check if the number you request has a match and ensures one match per number.

Reply:
Ok, so that key is not the formatted number as vcollantes said

Reply:
Yes, as I talked to @carlosfernandez and @jpmg93, a number can have multiple contacts,for this reason, instead of a map, the returned structure will be an array, and the phoneNumber will be inside.

Reply:
To confirm, is the key the number used to perform the request or a normalized number?

Reply:
The key was the normalizedNumber, please check the old structure (because now will be an array):
"payload":{ 
         "34600000001":{ 
            "firstName":"Josh",
            "middleName":"Claus",
            "lastName":"Snow",
            "encodedAvatar":"RXJlcyB1biBoYWNrZXI="
         },
         "34600000002":{ 
            "firstName":"Josh",
            "middleName":"Claus",
            "lastName":"Snow",
            "encodedAvatar":"RXJlcyB1biBoYWNrZXI="
         },
         "34600000003":{ 
            "firstName":"Josh",
            "middleName":"Claus",
            "lastName":"Snow",
            "encodedAvatar":"RXJlcyB1biBoYWNrZXI="
         },
         "34600000004":{ 
            "firstName":"Josh",
            "middleName":"Claus",
            "lastName":"Snow",
            "encodedAvatar":"RXJlcyB1biBoYWNrZXI="
         }
   }

where "34600000001" is the normalized number

Reply:
then, if the key is the normalized number, it cannot be used to check that the requested number has a match, as @jmpg93 said. That key should be the request number

Reply:
The requested number is the normalizedNumber, when I request the numbers, previosly I normalize those numbers, check the petition (it's an array of normalized numbers)

"type":"FETCH_CONTACT_DATA",
   "id":"5",
   "payload":{ 
      "numbers":[ 
         "34600000001",
         "34600000002",
         "34600000003",
         "34600000004"
      ]
   }

Reply:
So you are forcing the caller to implement a number normalization logic (which is not an easy task and probably requires libphonenumber)

I think it would be better to accept any kind of number (local ones and normalized) and leave the native side to perform the matching.

Keep in mind that this bridge can potentially be used by any OB, not only us

Reply:
That's a good point, but right now we only have one use case. In that use cases, the caller already has the normalized number, so it does not make sense to call numbering services again. 

If new use cases arise in the future, we can iterate over it.

Reply:
And the normalized number is served by backend, and backend already is using libphonenumber library, so there are no extra problems in this case.

Reply:
Thats true, at least in account, all normalization number are provided by server, avoiding including libphonenumber in client side, so, I would go with the current approach of having the keys as normalized numbers unless you think there will be another client of this bridge that wont have server providing these nombers

Reply:
Anyways, this code is a passthrough (this lib) from client code to native code and viceversa, if native part should normalize/denormalize numbers in the future, this code wont change.